test/integration/protocol_integration_test.cc

```diff
@@ -2687,4 +2687,40 @@ TEST_P(ProtocolIntegrationTest, ReqRespSizeStats) {
   test_server_->waitUntilHistogramHasSamples("cluster.cluster_0.upstream_rs_headers_size");
 }
 
+TEST_P(ProtocolIntegrationTest, ResetLargeResponseUponReceivingHeaders) {
```

Comment:
this can be DownstreamProtocolIntegrationTest no?

Reply:
This test talks to backend and it is interesting to cover H1 and H2 upstreams, so I would prefer it be ProtocolIntegrationTest